File: services/auth/seed.py
"""Crea el usuario Gerencia/Administrador inicial si no existe.

Uso: python seed.py
"""
import os

from sqlalchemy import select

from app.database import SessionLocal
from app.domain import EstadoUsuario, LineaNegocio, Rol
from app.models import User
from app.security import hash_password


def main() -> None:
    email = os.environ.get("SEED_ADMIN_EMAIL", "maya@grupomactral.com")
    password = os.environ.get("SEED_ADMIN_PASSWORD", "Mactral2026!")

    db = SessionLocal()
    try:
        existing = db.execute(select(User).where(User.email == email)).scalar_one_or_none()
        if existing:
            print(f"El usuario administrador {email} ya existe, no se crea de nuevo.")
            return

        user = User(
            name="Maya Lozada",
            email=email,
            password_hash=hash_password(password),
            role=Rol.GERENCIA,
            linea_negocio=LineaNegocio.AMBAS,
            status=EstadoUsuario.ACTIVO,
        )
        db.add(user)
        db.commit()
        print(f"Usuario administrador creado: {email} / {password}")
    finally:
        db.close()


if __name__ == "__main__":
    main()

File: prisma/seed.ts
import { PrismaClient } from "@prisma/client";
import bcrypt from "bcryptjs";

const prisma = new PrismaClient();

async function main() {
  const email = process.env.SEED_ADMIN_EMAIL ?? "maya@grupomactral.com";
  const password = process.env.SEED_ADMIN_PASSWORD ?? "Mactral2026!";

  const existing = await prisma.user.findUnique({ where: { email } });
  if (existing) {
    console.log(`El usuario administrador ${email} ya existe, no se crea de nuevo.`);
    return;
  }

  const passwordHash = await bcrypt.hash(password, 10);

  await prisma.user.create({
    data: {
      name: "Maya Lozada",
      email,
      passwordHash,
      role: "GERENCIA",
      lineaNegocio: "AMBAS",
      status: "ACTIVO",
    },
  });

  console.log(`Usuario administrador creado: ${email} / ${password}`);
}

main()
  .catch((error) => {
    console.error(error);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
